src/EnergyPlus/Plant/PlantManager.cc

```diff
@@ -3041,327 +3041,237 @@ void SizePlantLoop(EnergyPlusData &state,
+    auto &loop = state.dataPlnt->PlantLoop(LoopNum);
+    if (loop.PlantSizNum > 0) {
+        PlantSizNum = loop.PlantSizNum;
         // PlantSizData(PlantSizNum)%DesVolFlowRate = 0.0D0 ! DSU2
     } else {
         if (state.dataSize->NumPltSizInput > 0) {
-            PlantSizNum =
-                Util::FindItemInList(state.dataPlnt->PlantLoop(LoopNum).Name, state.dataSize->PlantSizData, &PlantSizingData::PlantLoopName);
+            PlantSizNum = Util::FindItemInList(loop.Name, state.dataSize->PlantSizData, &DataSizing::PlantSizingData::PlantLoopName);
         }
     }
-    state.dataPlnt->PlantLoop(LoopNum).PlantSizNum = PlantSizNum;
+    loop.PlantSizNum = PlantSizNum;
     // calculate a loop sizing factor and a branch sizing factor. Note that components without a sizing factor
     // are assigned sizing factors of zero in this calculation
     if (PlantSizNum > 0) {
+        auto &loopSizData = state.dataSize->PlantSizData(PlantSizNum);
         if (state.dataPlantMgr->GetCompSizFac) {
-            for (int BranchNum = 1; BranchNum <= state.dataPlnt->PlantLoop(LoopNum).LoopSide(LoopSideLocation::Supply).TotalBranches; ++BranchNum) {
-                BranchSizFac = 0.0;
-                state.dataPlnt->PlantLoop(LoopNum).LoopSide(LoopSideLocation::Supply).Branch(BranchNum).PumpSizFac = 1.0;
-                if (state.dataPlnt->PlantLoop(LoopNum).LoopSide(LoopSideLocation::Supply).NodeNumIn ==
-                    state.dataPlnt->PlantLoop(LoopNum).LoopSide(LoopSideLocation::Supply).Branch(BranchNum).NodeNumIn)
-                    continue;
-                if (state.dataPlnt->PlantLoop(LoopNum).LoopSide(LoopSideLocation::Supply).NodeNumOut ==
-                    state.dataPlnt->PlantLoop(LoopNum).LoopSide(LoopSideLocation::Supply).Branch(BranchNum).NodeNumOut)
-                    continue;
-                for (int CompNum = 1;
-                     CompNum <= state.dataPlnt->PlantLoop(LoopNum).LoopSide(LoopSideLocation::Supply).Branch(BranchNum).TotalComponents;
-                     ++CompNum) {
-                    state.dataPlnt->PlantLoop(LoopNum).LoopSide(LoopSideLocation::Supply).Branch(BranchNum).Comp(CompNum).simulate(state, true);
-                    BranchSizFac = max(BranchSizFac,
-                                       state.dataPlnt->PlantLoop(LoopNum).LoopSide(LoopSideLocation::Supply).Branch(BranchNum).Comp(CompNum).SizFac);
+            Real64 NumBrSizFac = 0.0;
+            Real64 LoopSizFac = 0.0;
+            Real64 MaxSizFac = 0.0;
+            auto &supplySide = loop.LoopSide(LoopSideLocation::Supply);
+            for (int BranchNum = 1; BranchNum <= supplySide.TotalBranches; ++BranchNum) {
+                auto &supplyBranch = supplySide.Branch(BranchNum);
+                Real64 BranchSizFac = 0.0;
+                supplyBranch.PumpSizFac = 1.0;
+                if (supplySide.NodeNumIn == supplyBranch.NodeNumIn) continue;
+                if (supplySide.NodeNumOut == supplyBranch.NodeNumOut) continue;
```

Comment:
This is interesting because this excludes the inlet and outlet branch from the BranchSizFac calculation and this was happening before. I know heat recovery HP companions are usually installed on a plant inlet and outlet branch so I wonder how this would affect sizing for that configuration? Nothing to do here just thinking about the implication.